### PR TITLE
feat: update doc for bitbucket cloud

### PR DESCRIPTION
**Pull request by likyh**
# Summary

add document for bitbucket

### CODE REVIEW

File: docs/Configuration/BitBucket.md [hunk 1]
```diff
@@ -47,20 +47,87 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
 
-
 ### Step 2 - Configure Blueprint
 
-Similar to other beta plugins, BitBucket does not support `project`, which means, you can only collect BitBucket data via blueprint's advanced mode. 
+![image](https://user-images.githubusercontent.com/3294100/220236338-772b30b7-974f-4bc2-89ce-f2abe5e92a5e.png)
+
+#### Repositories
+
+Select the Bitbucket repos to collect.
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: Can we use BitBucket instead of Bitbucket?

File: docs/Configuration/BitBucket.md [hunk 2]
```diff
@@ -47,20 +47,87 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+
+Without changing default transformation rules, you can still view the Metrics dashboard. However, if you want to view pre-built dashboards, the following transformation rules, especially "Type/Bug", should be added.<br/>
+
+Each Bitbucket repo has at most ONE set of transformation rules.
+
+#### Issue Tracking
+
+- TODO: the issues with selected states can be recognized not start issues. 
+
+- IN-PROGRESS: Same as "TODO".
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: IN-PROGRESS: The issues statuses that indicate an issue is work in progress
DONE: The issue statuses that indicate an issue is completed
OTHER: Other issues statuses that can not be mapped to the above three statuses

File: docs/Configuration/BitBucket.md [hunk 3]
```diff
@@ -47,20 +47,87 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+- DONE: Same as "TODO".
+- OTHER: Same as "TODO".
+
+#### CI/CD
+
+This set of configurations is used for calculating [DORA metrics](../DORA.md).
+
+A Bitbucket pipeline has many steps. Each Bitbucket pipeline is converted to a
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: BitBucket has several key CI entities: `pipelines`, `pipeline steps`, and `deployments`. A Bitbucket pipeline contains several pipeline steps. Each pipeline step defined with a deployment key can be mapped to a BitBucket deployment.

Each Bitbucket pipeline is converted to a cicd_pipeline in DevLake's domain layer schema and each Bitbucket pipeline step is converted to a cicd_task in DevLake's domain layer.

File: docs/Configuration/BitBucket.md [hunk 4]
```diff
@@ -47,20 +47,87 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+
+This set of configurations is used for calculating [DORA metrics](../DORA.md).
+
+A Bitbucket pipeline has many steps. Each Bitbucket pipeline is converted to a 
+cicd_pipeline in the domain layer and each Bitbucket pipeline step is converted to a cicd_task in the domain layer.
+![image](https://user-images.githubusercontent.com/3294100/220288225-71bee07d-c319-45bd-98e5-f4d01359840e.png)
+
+![image](https://user-images.githubusercontent.com/3294100/220289726-3909d368-1414-456c-a527-12a693745611.png)
+
+If you are using Bitbucket Deployments and Environments, please select "Detect Deployment from Environment in BitBucket". DevLake will consider the pipeline steps with deployment as a deployment and use the environment type of deployment in cicd_task.
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: I don't think there is a "Detect Deployment from Environment in BitBucket" in the Config UI.

**Review by Startrekzky — CHANGES_REQUESTED** [review]

File: docs/Configuration/BitBucket.md [hunk 5]
```diff
@@ -8,21 +8,21 @@ Visit config-ui: `http://localhost:4000` and go to `Connections` page.
...
+![image](https://user-images.githubusercontent.com/3294100/220118398-2d08070f-0edb-4de6-8696-9ee58b80719b.png)
 
 #### Connection Name
 
 Name your connection.
 
 #### Endpoint URL
 
-This should be a valid REST API endpoint for BitBucket: `https://api.bitbucket.org/2.0/`. The endpoint URL should end with `/`.
+This should be a valid REST API endpoint for BitBucket Cloud: `https://api.BitBucket.org/2.0/`. The endpoint URL should end with `/`.
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: This should be lowercase😂?

File: docs/Configuration/BitBucket.md [hunk 6]
```diff
@@ -8,21 +8,21 @@ Visit config-ui: `http://localhost:4000` and go to `Connections` page.
...
 
 DevLake will support BitBucket Server in the future.
 
 #### Authentication
 
-BitBucket `username` and `app password` are required to add a connection. Learn about [how to create a BitBucket app password](https://support.atlassian.com/bitbucket-cloud/docs/create-an-app-password/).
+BitBucket `username` and `app password` are required to add a connection. Learn about [how to create a BitBucket app password](https://support.atlassian.com/BitBucket-cloud/docs/create-an-app-password/).
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: So does this one.

File: docs/Configuration/BitBucket.md [hunk 7]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+
+#### Repositories
+
+Select the BitBucket repos to collect.
+
+#### Data Entities
+
+Usually, you don't have to modify this part. However, if you don't want to collect certain BitBucket entities, you can unselect some entities to accelerate the collection speed.
+
+- Issue Tracking: BitBucket issues, issue comments, etc.
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md:71]: Have we collected BitBucket issue comments? If not, delete it.

File: docs/Configuration/BitBucket.md [hunk 8]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+![image](https://user-images.githubusercontent.com/3294100/220338276-a67cd8cc-ea76-4cb2-bb7b-bba581d21d70.png)
+
+Without changing default transformation rules, you can still view the Metrics dashboard. However, if you want to view pre-built dashboards, the following transformation rules, especially "Type/Bug", should be added.<br/>
+
+Each BitBucket repo has at most ONE set of transformation rules.
+
+#### Issue Tracking
+
+- TODO: the issues with selected states can be recognized not start issues. 
+
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: No blank line here.

File: docs/Configuration/BitBucket.md [hunk 9]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+- CI/CD: BitBucket Pipelines, BitBucket Deployments, etc.
+- Cross Domain: BitBucket accounts, etc.
 
 Please go to the `Blueprints` page and switch to advanced mode. See how to use advanced mode and JSON [examples](AdvancedMode.md).
 
+### Step 3 - Adding Transformation Rules (Optional)
+
+![image](https://user-images.githubusercontent.com/3294100/220338276-a67cd8cc-ea76-4cb2-bb7b-bba581d21d70.png)
+
+Without changing default transformation rules, you can still view the Metrics dashboard. However, if you want to view pre-built dashboards, the following transformation rules, especially "Type/Bug", should be added.<br/>
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: Why here is a `especially "Type/Bug", should be added.`?

File: docs/Configuration/BitBucket.md [hunk 10]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+#### CI/CD
+
+This set of configurations is used for calculating [DORA metrics](../DORA.md).
+
+BitBucket has several key CI entities: `pipelines`, `pipeline steps`, and `deployments`. A Bitbucket pipeline contains several pipeline steps. Each pipeline step defined with a deployment key can be mapped to a BitBucket deployment.
+
+Each Bitbucket pipeline is converted to a cicd_pipeline in DevLake's domain layer schema and each Bitbucket pipeline step is converted to a cicd_task in DevLake's domain layer.
+![image](https://user-images.githubusercontent.com/3294100/220288225-71bee07d-c319-45bd-98e5-f4d01359840e.png)
+
+![image](https://user-images.githubusercontent.com/3294100/220289726-3909d368-1414-456c-a527-12a693745611.png)
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: 1. Annotation change: If a pipeline step defines `deployment` with a value (usually means envrionment), this pipeline step is also a BitBucket deployment. Their relationship is 1:1
image

2. Add a screenshot of how to define deployment and environment in `.yaml` to explain point 1. For example,
1g9BrGdVlb

File: docs/Configuration/BitBucket.md [hunk 11]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+
+If you are using BitBucket Deployments and Environments, DevLake will consider the pipeline steps with deployment as a deployment and use the environment type of deployment in cicd_task.
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md]: How does DevLake tell if a BitBucket pipeline step is a deployment? The pipeline steps (defined in the .yaml) with the `deployment` key are considered as `DevLake deployments`. The value of the `deployment` key will be considered as the environment of DevLake deployments.

These DevLake deployments will be recorded in table.cicd_tasks in DevLake's domain layer, with `type` = 'deployment' and `environment` = '{BitBucket-pipeline-step.deployment.value}', differentiating from other CI tasks.

File: docs/Configuration/BitBucket.md [hunk 12]
```diff
@@ -47,20 +47,88 @@ If you are behind a corporate firewall or VPN you may need to utilize a proxy se
...
+
+Or if you're using BitBucket pipelines to conduct `deployments`, please select "Detect Deployments from Pipeline steps in BitBucket", and input the RegEx in the following fields:
```
Comment (Startrekzky) [inline docs/Configuration/BitBucket.md:117]: The following paragraphs should be updated according to BitBucket's transformation page as below.

image

image

**Review by Startrekzky — CHANGES_REQUESTED** [review]

File: docs/Configuration/BitBucket.md (hunk 7 above)
Comment (likyh) [reply docs/Configuration/BitBucket.md:71]: Yes, 
image

File: docs/Configuration/BitBucket.md (hunk 10 above)
Comment (likyh) [reply docs/Configuration/BitBucket.md]: I add 1 in doc and add a notice in pic.

**Review by Startrekzky — APPROVED** [review]
I'll merge it first and @yumengwang03 can you help refine the wording and syntax?

**Pull request merged**